utils/config/config.go

```diff
@@ -0,0 +1,168 @@
+package config
+
+import (
+	"errors"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+)
+
+type Config struct {
+	*viper.Viper
+}
+
+var v = viper.New()
+
+func SetOptionalsDefault(grantType, scope string, expiration int) (*Config, error) {
+	viper.Set("grant_type", grantType)
+	viper.Set("scope", scope)
+	viper.Set("expiration", expiration)
+
+	return &Config{viper.GetViper()}, nil
+}
+
+func WriteCredentials(credendialsFile, clientId, clientSecret, accountId, accountEnvId string) (*Config, error) {
+	homeDir, err := os.UserHomeDir()
+	cobra.CheckErr(err)
+
+	if _, err := os.Stat(homeDir + "/.flagship"); errors.Is(err, os.ErrNotExist) {
+		err := os.Mkdir(homeDir+"/.flagship", os.ModePerm)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	filepath, _ := filepath.Abs(homeDir + "/.flagship/" + credendialsFile)
+	v.SetConfigFile(filepath)
+	v.Set("client_id", clientId)
+	v.Set("client_secret", clientSecret)
+	v.Set("account_id", accountId)
+	v.Set("account_environment_id", accountEnvId)
+	err = v.WriteConfigAs(filepath)
+	if err != nil {
+		log.Fatalf("error occured: %v", err)
+	}
+
+	return &Config{v}, nil
+
+}
+
+func WriteOptionals(credendialsFile, grantType, scope string, expiration int) (*Config, error) {
+	homeDir, err := os.UserHomeDir()
+	cobra.CheckErr(err)
+
+	if _, err := os.Stat(homeDir + "/.flagship"); errors.Is(err, os.ErrNotExist) {
```

Comment:
Could you refactor those lines, they are strictly identical to the ones in WriteCredentials I think.